be/src/olap/segment_loader.cpp

```diff
@@ -59,8 +59,14 @@ Status SegmentLoader::load_segments(const BetaRowsetSharedPtr& rowset,
     for (int64_t i = 0; i < rowset->num_segments(); i++) {
         SegmentCache::CacheKey cache_key(rowset->rowset_id(), i);
         if (_segment_cache->lookup(cache_key, cache_handle)) {
-            continue;
+            // Has to check the segment status here, because the segment in cache may has something wrong during
```

Comment:
Why not check status in lookup and erase not healthy segment?

Reply:
because the segment cache code is too complicated, it may produce memory leak issue during lookup.